pkg/apiserver/certificate/selfsignedcert_provider_test.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certificate
+
+import (
+	"context"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	genericoptions "k8s.io/apiserver/pkg/server/options"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	certutil "k8s.io/client-go/util/cert"
+	clocktesting "k8s.io/utils/clock/testing"
+
+	"antrea.io/antrea/pkg/util/env"
+)
+
+const (
+	testServiceName = "svc-foo"
+	testPairName    = "foo"
+	testSecretName  = "secret-foo"
+)
+
+var (
+	// one year self-signed certs.
+	testOneYearCert, testOneYearKey, _   = certutil.GenerateSelfSignedCertKeyWithFixtures("localhost", loopbackAddresses, nil, "")
+	testOneYearCert2, testOneYearKey2, _ = certutil.GenerateSelfSignedCertKeyWithFixtures("localhost", loopbackAddresses, nil, "")
+	testOneYearCert3, testOneYearKey3, _ = certutil.GenerateSelfSignedCertKeyWithFixtures("localhost", loopbackAddresses, nil, "")
+)
+
+func newTestSelfSignedCertProvider(t *testing.T, client *fakeclientset.Clientset, tlsSecretName string, maxRotateDuration time.Duration) *selfSignedCertProvider {
+	secureServing := genericoptions.NewSecureServingOptions().WithLoopback()
+	caConfig := &CAConfig{
+		TLSSecretName:         tlsSecretName,
+		SelfSignedCertDir:     t.TempDir(),
+		MinValidationDuration: maxRotateDuration,
```

Comment:
you're using the `maxRotateDuration` term here still, yet you use the parameter value to initialize `MinValidationDuration`. But the 2 mean different things right?